### PR TITLE
BigInt

### PR DESCRIPTION
**Pull request by danilolmc**

### CODE REVIEW

**Review by nazarepiedady — COMMENTED** [review]
@danilolmc, what do you think about consider the idea about translating the word `bigint` in the article content as `número inteiro grande` or `inteiro grande` or `inteiro maior`?

@Peruibeloko, @jonnathan-ls, @odsantos, what do you think about the suggestion I did?

**Comment by Peruibeloko** [comment]
@nazarepiedady I'd keep it as BigInt, and the only change I'd suggest is to capitalize it, since we're dealing with the actual object's name, much like we shouldn't translate `Promise` to `Promessa` for example :)

**Comment by danilolmc** [comment]
I agree with @Peruibeloko, since BigInt is just like Promise and a data type itself.

@Peruibeloko do you mean capitalize every bigint word in the article? I just let some bigint words as it was already capitalized.

**Comment by Peruibeloko** [comment]
@danilolmc Yes, that would be the case. Just a suggestion, of course.

**Comment by danilolmc** [comment]
Ok, no problem, I'm going to capitalize them!

**Comment by nazarepiedady** [comment]
> @nazarepiedady I'd keep it as BigInt, and the only change I'd suggest is to capitalize it, since we're dealing with the actual object's name, much like we shouldn't translate `Promise` to `Promessa` for example :)

@Peruibeloko, I was talking about to translate the usage of the word **bigint** where it is not reference the data type.

**Comment by Peruibeloko** [comment]
@nazarepiedady the problem is that `bigint` always refers to the data type

File: 1-js/99-js-misc/05-bigint/article.md [hunk 1]
```diff
@@ -1,130 +1,129 @@
 # BigInt
 
-[recent caniuse="bigint"]
+[BigInt recentemente adicionado ao caniuse="bigint"]
 
-`BigInt` is a special numeric type that provides support for integers of arbitrary length.
+BigInt é um tipo numérico especial que oferece suporte para inteiros de comprimento arbitrário.
```
Comment (Peruibeloko) [inline 1-js/99-js-misc/05-bigint/article.md:5]: Eu usaria o termo "tamanho" ao invés de comprimento

File: 1-js/99-js-misc/05-bigint/article.md [hunk 2]
```diff
@@ -1,130 +1,129 @@
...
 
-const bigintFromNumber = BigInt(10); // same as 10n
+const bigintFromNumber = BigInt(10); // o mesmo que 10n
 ```
 
-## Math operators
+## Operadores matemáticos
 
-`BigInt` can mostly be used like a regular number, for example:
+`BigInt` pode ser usado principalmente como um número regular, por exemplo:
```
Comment (Peruibeloko) [inline 1-js/99-js-misc/05-bigint/article.md:19]: O termo "Regular" em inglês é melhor traduzido para "comum" nesses contextos

File: 1-js/99-js-misc/05-bigint/article.md [hunk 3]
```diff
@@ -1,130 +1,129 @@
...
 alert(1n + 2n); // 3
 
 alert(5n / 2n); // 2
 ```
 
-Please note: the division `5/2` returns the result rounded towards zero, without the decimal part. All operations on bigints return bigints.
+Perceba: a divisão `5/2` retorna o resultado arredondado para zero, sem a parte decimal. Todas as operações com bigints retornam bigints.
 
-We can't mix bigints and regular numbers:
+Não podemos misturar bigints e números regulares:
```
Comment (Peruibeloko) [inline 1-js/99-js-misc/05-bigint/article.md:30]: Regular -> Comum

File: 1-js/99-js-misc/05-bigint/article.md [hunk 4]
```diff
@@ -1,130 +1,129 @@
...
 
 ```js run
-alert(1n + 2); // Error: Cannot mix BigInt and other types
+alert(1n + 2); // Erro: Não é possível misturar BigInt e outros tipos
 ```
 
-We should explicitly convert them if needed: using either `BigInt()` or `Number()`, like this:
+Devemos convertê-los explicitamente se necessário: usando `BigInt()` ou `Number()`, assim:
```
Comment (Peruibeloko) [inline 1-js/99-js-misc/05-bigint/article.md:35]: Eu costumo remover marcações de `:` que eu considero desnecessárias ou gramaticalmente incorretas em PT. Aqui por exemplo, eu usaria "[...] explicitamente se necessário, usando [...]"

A vírgula oferece uma continuidade melhor ao texto

File: 1-js/99-js-misc/05-bigint/article.md [hunk 5]
```diff
@@ -1,130 +1,129 @@
...
+// bigint para number
 alert(Number(bigint) + number); // 3
 ```
 
-The conversion operations are always silent, never give errors, but if the bigint is too huge and won't fit the number type, then extra bits will be cut off, so we should be careful doing such conversion.
+As operações de conversão são sempre silenciosas, nunca dão erros, mas se o bigint for muito grande e não couber no tipo número, então bits extras serão cortados, então devemos ter cuidado ao fazer a conversão.
 
-````smart header="The unary plus is not supported on bigints"
-The unary plus operator `+value` is a well-known way to convert `value` to a number.
+````smart header="O operador unário mais não é suportado em bigints"
```
Comment (Peruibeloko) [inline 1-js/99-js-misc/05-bigint/article.md:50]: A ordem no português é invertida, fica "mais unário", já que "unário" é um adjetivo de "mais"

File: 1-js/99-js-misc/05-bigint/article.md [hunk 6]
```diff
@@ -1,130 +1,129 @@
...
+alert( 1n || 2 ); // 1 (1n é considerado verdadeiro)
 
-alert( 0n || 2 ); // 2 (0n is considered falsy)
+alert( 0n || 2 ); // 2 (0n é considerado falso)
 ```
 
 ## Polyfills
 
-Polyfilling bigints is tricky. The reason is that many JavaScript operators, such as `+`, `-` and so on behave differently with bigints compared to regular numbers.
+Fazer polyfill para bigints é complicado. A razão é que muitos operadores JavaScript, como `+`, `-` e assim por diante se comportam de maneira diferente com bigints em comparação com números regulares.
```
Comment (Peruibeloko) [inline 1-js/99-js-misc/05-bigint/article.md:101]: Polyfill na tradução se torna um substantivo, ficaria então "Fazer um polyfill" ou "Fazer polyfills", eu prefiro o primeiro

File: 1-js/99-js-misc/05-bigint/article.md [hunk 7]
```diff
@@ -1,130 +1,129 @@
...
+Por exemplo, a divisão de bigints sempre retorna um bigint (arredondado se necessário).
 
-To emulate such behavior, a polyfill would need to analyze the code and replace all such operators with its functions. But doing so is cumbersome and would cost a lot of performance.
+Para emular tal comportamento, um polyfill precisaria analisar o código e substituir todos esses operadores com suas funções. Mas fazer isso é trabalhoso e custaria muito em termos de desempenho.
 
-So, there's no well-known good polyfill.
+Então, não há um polyfill bem conhecido e bom.
 
-Although, the other way around is proposed by the developers of [JSBI](https://github.com/GoogleChromeLabs/jsbi) library.
+Embora, o caminho inverso é proposto pelos desenvolvedores da biblioteca [JSBI](https://github.com/GoogleChromeLabs/jsbi).
```
Comment (Peruibeloko) [inline 1-js/99-js-misc/05-bigint/article.md:109]: "Embora" me parece meio desconexo aqui, que tal "Apesar disso"?

File: 1-js/99-js-misc/05-bigint/article.md [hunk 8]
```diff
@@ -1,130 +1,129 @@
...
 
-We can use such JSBI code "as is" for engines that don't support bigints and for those that do support - the polyfill will convert the calls to native bigints.
+Podemos usar esse código JSBI "como está" para motores que não suportam bigints e para aqueles que suportam - o polyfill converterá as chamadas para bigints nativos.
 
-## References
+## Referências
 
-- [MDN docs on BigInt](mdn:/JavaScript/Reference/Global_Objects/BigInt).
-- [Specification](https://tc39.es/ecma262/#sec-bigint-objects).
+- [Documentação da MDN sobre BigInt](mdn:/JavaScript/Reference/Global_Objects/BigInt).
```
Comment (Peruibeloko) [inline 1-js/99-js-misc/05-bigint/article.md:128]: Acho que vale usar o link para a MDN em PT

**Review by Peruibeloko — CHANGES_REQUESTED** [review]
Solicitei algumas mudanças, deve ser rápido

**Review by Peruibeloko — APPROVED** [review]

**Comment by odsantos** [comment]
> I agree with @Peruibeloko, since BigInt is just like Promise and the data type name itself.
> 
> @Peruibeloko do you mean capitalize every bigint word in the article? I just let some bigint words as it was already capitalized.

Seems good, as the Portuguese 'mdn' link, and other accepted suggestions here.

**Review by nazarepiedady — APPROVED** [review]

**Pull request merged**